### ISSUE TITLE
PetStore test server returns wrong data format for loginUser call

### ISSUE THREAD

**Issue by evigeant** [swagger-api/swagger-codegen#1660]
I am preparing a PR related to cleaning up the Java Jersey client. In the process, I have rebuilt the PetStore Java client samples.

But while running the JUnit tests against the PetStore server(http://petstore.swagger.io/v2), I get an error on the `UserApi.loginUser` call.
The swagger API spec states that it should produce:
- `application/json`
- `application/xml`

In the response headers, `application/json` is specified, but I received:

    logged in user session:...

This is not valid JSON, so my new code fails to deserialize it. However, I see in the old code (before my  modifications) that there is a try catch and an exception for the String case.

Is it possible to do one of the following:
- Modify the test server to send valid JSON
- Change the spec and the response headers to indicate `text/plain`

**Comment by wing328** [comment]
@evigeant I agree with you that applicaiton/json is not the correct content type for the response and I prefer your second suggestion to change the content-type of the response to text/plain so that we have another test case to cover response content-type other than application/json and application/xml.

**Comment by evigeant** [comment]
@wing328 indeed, I think it a good idea to test another content-type, however this will require changing the petstore spec (it currently states that it produces `application/json` or `application/xml`) and this might break the automated tests for some languages. I haven't checked, but I have seen that there are a lot of generators.

Also, if you are going to change the spec of the PetStore sample, I would consider fixing #1661 at the same time.
